src/inputs/TextField/index.tsx

```diff
@@ -1,25 +1,35 @@
 import React from 'react';
 import TextFieldMui from '@material-ui/core/TextField';
-//import { withStyles } from '@material-ui/core/styles';
+import InputAdornment from '@material-ui/core/InputAdornment';
 import styled from 'styled-components';
 
-//import theme from '../../theme';
-
 type Props = {
   value: string;
   onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
   label: string;
   readOnly?: boolean;
   meta?: any;
   input?: any; // added for compatibility with react-final-form
+  startAdornment?: React.ReactElement;
+  endAdornment?: React.ReactElement;
 } & React.InputHTMLAttributes<HTMLInputElement>;
 
-const CustomTextField = styled(({ input, ...props }) => (
-  <TextFieldMui {...props} />
-))<Props>`
+const CustomTextField = styled(
+  ({ input, startAdornment, endAdornment, ...props }) => (
```

Comment:
Are these typed?

Reply:
I did some changes, now it only receives `...props` but it's a little tricky to type it. It Should be typed to `TextFieldProps` but TS is complaining about some overloads props and the message is not friendly at all. 

I think it's having problems with InputProps args, but not I'm not sure, could you take a look, please?

Reply:
this is very hard to type, please don't use spread operators that much. Also, keep in mind that most of our components are just wrappers over Material UI components and they have their prop types exported (for example there is TextFieldProps) and we may take use of it

Reply:
Agreed, but in this particular case, we need to propagate the input props. How could we do it without using spread?

Reply:
I'm not saying anything about this particular case but in general